administration/src/bp-modules/cards/ImportCardsRequirementsText.tsx

```diff
@@ -18,15 +19,31 @@ const ImportCardsRequirementsText = ({
   csvHeaders,
   isFreinetFormat = false,
 }: ImportCardsRequirementsProps): ReactElement => {
+  const projectConfig = useContext(ProjectConfigContext)
+  const { card: cardConfig } = projectConfig
   const { t } = useTranslation('cards')
+
+  const requiredHeaders = useMemo(() => {
+    const headers = [cardConfig.nameColumnName, cardConfig.expiryColumnName]
+    cardConfig.extensions.forEach((extension, index) => {
+      const header = cardConfig.extensionColumnNames[index]
+      if (extension.isMandatory && header) {
+        headers.push(header)
+      }
+    })
+    return headers
+  }, [cardConfig])
```

Comment:
A functional approach (which I find easier to understand) would be this:
```suggestion
  const requiredHeaders = useMemo(
    () => [
      projectConfig.card.nameColumnName,
      projectConfig.card.expiryColumnName,
      ...projectConfig.card.extensions.flatMap((extension, index) => {
        const header = projectConfig.card.extensionColumnNames[index]
        return extension.isMandatory && header ? [header] : []
      }),
    ],
    [projectConfig.card]
  )
```

Also I think we re overusing `useMemo()` quiet a bit in our code base. For example, here it is quiet unnecessary, since `cardConfig` changes almost never in practical use. We could use a plain function here.

Reply:
i agree. Even there is not state in this function it could also be outside of the component to avoid unneeded rerendering. But just removing `useMemo` should be sufficient